### ISSUE TITLE
Node.js Technical Steering Committee (TSC) Meeting 2023-09-13

### ISSUE THREAD

**Issue by mhdawson** [nodejs/TSC#1437]
## Time

**UTC Wed 13-Sep-2023 15:00 (03:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Wed 13-Sep-2023 08:00 (08:00 AM) |
| US / Mountain | Wed 13-Sep-2023 09:00 (09:00 AM) |
| US / Central  | Wed 13-Sep-2023 10:00 (10:00 AM) |
| US / Eastern  | Wed 13-Sep-2023 11:00 (11:00 AM) |
| EU / Western  | Wed 13-Sep-2023 16:00 (04:00 PM) |
| EU / Central  | Wed 13-Sep-2023 17:00 (05:00 PM) |
| EU / Eastern  | Wed 13-Sep-2023 18:00 (06:00 PM) |
| Moscow        | Wed 13-Sep-2023 18:00 (06:00 PM) |
| Chennai       | Wed 13-Sep-2023 20:30 (08:30 PM) |
| Hangzhou      | Wed 13-Sep-2023 23:00 (11:00 PM) |
| Tokyo         | Thu 14-Sep-2023 00:00 (12:00 AM) |
| Sydney        | Thu 14-Sep-2023 01:00 (01:00 AM) |

Or in your local time:
* https://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Technical%20Steering%20Committee%20(TSC)+Meeting+2023-09-13&iso=20230913T1500
* or https://www.wolframalpha.com/input/?i=03PM+UTC%2C+Sep+13%2C+2023+in+local+time

## Links

* Minutes Google Doc: <https://docs.google.com/document/d/116gcroy3PK_W3lRIaufWMmCCwvjpqOYAxM-fdImWhcs/edit>

## Agenda

Extracted from **tsc-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/node

* websocket: proposal for a new core module [#49478](https://github.com/nodejs/node/pull/49478)
* Discussion: New “ESM by default” mode [#49432](https://github.com/nodejs/node/issues/49432)

### nodejs/citgm

* lookup: declare bankruptcy on flaky modules [#959](https://github.com/nodejs/citgm/pull/959)

### nodejs/TSC

* Additions to `import.meta` [#1430](https://github.com/nodejs/TSC/issues/1430)

### nodejs/undici

* expose websocket in node bundle [#2217](https://github.com/nodejs/undici/pull/2217)

### nodejs/uvwasi

* Next release [#228](https://github.com/nodejs/uvwasi/issues/228)

## Invited

* Antoine du Hamel @aduh95 (voting member)
* Yagiz Nizipli @anonrig (voting member)
* Anatoli Papirovski @apapirovski (voting member)
* Benjamin Gruenbaum @benjamingr (voting member)
* Ruben Bridgewater @BridgeAR (voting member)
* Colin Ihrig @cjihrig (voting member)
* Danielle Adams @danielleadams (voting member)
* Geoffrey Booth @GeoffreyBooth (voting member)
* Gireesh Punathil @gireeshpunathil (voting member)
* James Snell @jasnell (voting member)
* Joyee Cheung @joyeecheung (voting member)
* Chengzhong Wu @legendecas (voting member)
* Matteo Collina @mcollina (voting member)
* Michael Dawson @mhdawson (voting member)
* Moshe Atlow @MoLow (voting member)
* Rafael Gonzaga @RafaelGSS (voting member)
* Darshan Sen @RaisinTen (voting member)
* Richard Lau @richardlau (voting member)
* Robert Nagy @ronag (voting member)
* Ruy Adorno @ruyadorno (voting member)
* Michaël Zasso @targos (voting member)
* Tobias Nießen @tniessen (voting member)
* Rich Trott @Trott (voting member)
* Beth Griggs @BethGriggs (regular member)
* Ben Noordhuis @bnoordhuis (regular member)
* Сковорода Никита Андреевич @ChALkeR (regular member)
* Shelley Vohr @codebytere (regular member)
* Daniel Bevenius @danbev (regular member)
* Franziska Hinkelmann @fhinkel (regular member)
* Gabriel Schulhof @gabrielschulhof (regular member)
* Brian White @mscdex (regular member)
* Myles Borins @MylesBorins (regular member)
* Rod Vagg @rvagg (regular member)
* Tiancheng Timothy Gu @TimothyGu (regular member)

## Notes

The agenda comes from issues labelled with `tsc-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

Zoom link: <https://zoom.us/j/611357642>
Regular password

## Public participation

We stream our conference call straight to YouTube so anyone can listen to it live, it should start playing at **<https://www.youtube.com/c/nodejs+foundation/live>** when we turn it on. There's usually a short cat-herding time at the start of the meeting and then occasionally we have some quick private business to attend to before we can start recording & streaming. So be patient and it should show up.

---

**Invitees**

Please use the following emoji reactions in this post to indicate your
availability.

* :+1: - Attending
* :-1: - Not attending
* :confused: - Not sure yet

**Comment by mhdawson** [comment]
Removing `Next release https://github.com/nodejs/uvwasi/issues/228` as I'd put the wrong label on the issue.

**Comment by GeoffreyBooth** [comment]
> Should we skip?

No, we need to unblock https://github.com/nodejs/citgm/pull/959, unless you’d like to remove your block on that.

I’d also like to discuss https://github.com/nodejs/node/issues/49432. I want to make it happen before 21.0.0 and there aren’t many more weeks left before that, so if the TSC has any concerns with what I’m aiming to do then it would be better to hear those sooner than later.

**Comment by anonrig** [comment]
I'd also like to discuss https://github.com/nodejs/TSC/issues/1438

**Comment by mcollina** [comment]
We should really cover https://github.com/nodejs/node/issues/49625

**Comment by GeoffreyBooth** [comment]
Questions for the TSC to answer re https://github.com/nodejs/node/issues/49432:

- Does anyone have any concerns about the proposal outlined in the top post at https://github.com/nodejs/node/issues/49432#issue-1877773681? In particular, any concerns that would motivate a block on a PR that implements what’s proposed here.

- It was suggested that rather than creating a new flag for this, we expand the scope of `--input-type`—instead of only affecting STDIN, `--eval` and `--print` it would enable _all_ forms of input including files to be treated as ESM by default. Leaving aside what flag name we’d use for the experimental phase, once we think the feature is stable [would it be considered a breaking/semver-major change](https://github.com/nodejs/node/issues/49541#issuecomment-1715013905) to expand the behaviors modified by `--input-type`? If that’s a change that we shipped, could it be semver-minor that gets backported?

**Comment by mhdawson** [comment]
If we can move to the 3-4 timeslot I can chair.  If we have another volunteer to chair at the current time we can leave as is. Maybe @gireeshpunathil or @Trott ?

I'll check first thing my morning on Wednesday. If there is a volunteer we'll leave as is, otherwise I'll move to the new time.

**Comment by mcollina** [comment]
I will make myself available to chair.

**Comment by mhdawson** [comment]
@mcollina thanks.

**Comment by Trott** [comment]
I'm the one who added the CITGM issue to the agenda, but I am very unlikely to be able to make the meeting this week. I'd say that for now, if we're chipping away at the module failures, we don't need to make any imminent decisions on skipping/removing modules. It's only once everything grinds to a halt again that we'll want to assess where we're at and what (if anything) to do.

**Comment by Trott** [comment]
> I'm the one who added the CITGM issue to the agenda, but I am very unlikely to be able to make the meeting this week. I'd say that for now, if we're chipping away at the module failures, we don't need to make any imminent decisions on skipping/removing modules. It's only once everything grinds to a halt again that we'll want to assess where we're at and what (if anything) to do.

Oh, looks like consensus has been reached there. I'll remove the label from that pull request.

**Comment by jasnell** [comment]
For CITGM, the consensus on the call was to trim the list of modules that CITGM handles to make things easier for now. Then, longer term, spin up a strategic initiative (champion to be determined) to improve CITGM overall.

**Comment by mhdawson** [comment]
@mcollina were you planning to PR in minutes for this meeting?
